js/page-specific/trails.js

```diff
@@ -1,907 +1,982 @@
-    };
-
-    window.myData = appData;
-
-    //Hide the initial message that describes a list of road closures should exist here
-    $('.initial-message').hide();
-
-    //HACK: To fix issue when going back to the app with a filter applied for demo
-    //This will be udpated in a upcoming release
-    var $filter = $('.js-trails-filter');
-    if ($filter.val()) {
-        $filter.val('');
-        $('body').focus();
-    }
-
-    map = new Map("map", {
-        basemap: "hybrid",
-        center: [-76.6455, 39.4671],
-        zoom: 10,
-        displayGraphicsOnPan: false,
-        autoResize: false
-    });
-
-    map.on("load", function () {
-
-        //On Zoom Change
-        map.on("extent-change", onZoom);
-
-        parkLayer = new FeatureLayer("//gis.baltimorecountymd.gov/arcgis/rest/services/Apps/WalkingTrailsExternalViewer/MapServer/0", {
-            name: "parks",
-            mode: esri.layers.FeatureLayer.MODE_SNAPSHOT,
-            outFields: ["*"], //Return all fields from the layer
-            infoTemplate: new InfoTemplate("") //This can be empty because we are going to update it later in the app
-        });
+        var urls = {
+            parkLayer: '//bcgis.baltimorecountymd.gov/arcgis/rest/services/Apps/WalkingTrails/MapServer/0',
+            trailLayer: '//bcgis.baltimorecountymd.gov/arcgis/rest/services/Apps/WalkingTrails/MapServer/1'
+        };
 
-        //Get a list of parks and display them in a list
-        parkLayer.on("load", GetParksList);
+        var parkInfoWindowTemplate,
+            trailsTemplate,
+            trailTemplate,
+            attachTemplate,
+            map,
+            parkLayer,
+            trailLayer,
+            pageInfo,
+            pageLoaded = false,
+            appData = {
+                ids: [],
+                parks: [],
+                trails: [],
+                attachs: []
+            },
+            coalitionMessage = "<a href='/Agencies/health/coalition/lhc.html' title='Learn more about the Baltimore County Health Colation'><img src='/sebin/e/t/coalitionapproved.gif' alt='This walking trail location is Health Coalition approved.' /></a>";
+
+        parkInfoWindowTemplate = function (desc) {
+            return "<div id='info' class='park-info'><p>" + desc + "</p>" +
+                "<a href='https://www.google.com/maps/?daddr=${Lat},${Long}'>" +
+                "Get Directions</a><a href='#parkid${ID}-${OBJECTID}' class='js-infoWindow-more-info pull-right' " +
+                "data-object-id='${OBJECTID}' title='View More Information on ${NAME}'>More Information</a>" +
+                "<div id='attach'></div>" +
+                "</div>";
+        };
 
-        //Listen for click event on point
-        parkLayer.on("click", DisplayParkInfoWindow);
+        trailsTemplate = "<ul class='trails'>{{#.}}{{#attributes}}{{#NAME}}";
```

Comment:
Would you consider creating this template in a HTML file, and then loading it into the DOM via ajax?